src/3rd_party_config.c

```diff
@@ -0,0 +1,235 @@
+/*
+ *   Software Updater - client side
+ *
+ *      Copyright © 2019 Intel Corporation.
+ *
+ *   This program is free software: you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation, version 2 or later of the License.
+ *
+ *   This program is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "3rd_party_internal.h"
+#include "config.h"
+#include "lib/config_parser.h"
+#include "lib/list.h"
+#include "lib/sys.h"
+#include "swupd.h"
+#include <errno.h>
+#include <stdlib.h>
+
+#ifdef THIRDPARTY
+#define MAX_REPO_ALLOWED_FIELDS 2
+
+/** @brief Internal Memory representation of a repo.  */
+struct repo {
+	char *name;
+	char *url;
+};
+
+/** @brief List of repos. */
+struct list *repos_kv;
+
+/**
+ * @brief This function is called by the repo ini parser It is called
+
+ *
+ * @param section A string containing section of the file
+ * @param key A string containing key
+ * @param value A string containing value
+ *
+ * @returns True for every valid line which could be extracted, false if either
+ * section, key or value is NULL
+ */
+bool parse_key_values(char *section, char *key, char *value, void UNUSED_PARAM *data)
+{
+	static int repo_field_count = 0;
+	bool ret = false;
+
+	if (section && key && value) {
+		static struct repo *repo_var;
+		if (repo_field_count == 0) {
+			repo_var = (struct repo *)malloc(sizeof(struct repo));
+			if (repo_var == NULL) {
+				error("Failed to allocate memory");
+				exit(1);
+			}
+
+			repo_var->name = NULL;
+		}
+
+		/* Section name is the name of the repo and needs to only parsed
+		 * the first time
+		 */
+		if (!repo_var->name) {
+			repo_var->name = strdup_or_die(section);
+			repo_field_count++;
+			ret = true;
+		}
+
+		if (!strncmp(key, "url", strlen(key))) {
+			repo_var->url = strdup_or_die(value);
+			repo_field_count++;
+			ret = true;
+		}
+
+		if (repo_field_count == MAX_REPO_ALLOWED_FIELDS) {
+			repos_kv = list_append_data(repos_kv, repo_var);
+			repo_field_count = 0;
+		}
+	}
+	return ret;
+}
+
+static char *get_repo_config_path(void)
+{
+	char *repo_config_file_path;
+	string_or_die(&repo_config_file_path, "%s/%s/%s", globals.state_dir, "3rd_party", "repo.ini");
+	return repo_config_file_path;
+}
+
+static int write_config_repo(struct repo *repo, int mode)
+{
+	int ret = 0;
+	char *repo_string;
+	ssize_t write_line_len;
+	char *repo_config_file_path = get_repo_config_path();
+
+	int fd = open(repo_config_file_path, mode);
+	if (fd < 0) {
+		return -1;
+	}
+
+	string_or_die(&repo_string, "[%s]\nurl=%s\n\n", repo->name, repo->url);
```

Comment:
I would add a n extra line between the section and key=value:
```
[MY_SECTION]

url=/some/url
```

Reply:
Thanks will change in next push